src/lib/client.ts

```diff
@@ -9,6 +9,7 @@ import { ShareResource } from './resources/share.resource';
 export interface PermanentConstructorConfigI {
   sessionToken: string;
   mfaToken: string;
+  archiveId: number;
```

Comment:
The contents of this interface are all required at construction; how are clients intended to find the `archiveId`?

Reply:
I think this is more clear in #18 and it looks like you added a commit about it.  I mentioned this in slack but review will probably be easier if you/we start with that one, since #19 and this one depend on it.

Reply:
Ah, the commit I added in 18 was about making `archiveNbr` optional; sounds like the same solution should apply to `archiveId`?

Reply:
Made this optional!